lib/broccoli_sass_compiler.js

```diff
@@ -273,12 +274,23 @@ BroccoliSassCompiler.prototype.logCompilationFailure = function(details, error)
 
 BroccoliSassCompiler.prototype.compileTree = function(srcPath, files, destDir) {
   var self = this;
-
-  return files.reduce(function(promise, sassFile) {
-    return promise.then(function() {
-      return RSVP.all(self.compileSassFile(srcPath, sassFile, destDir));
-    });
-  }, RSVP.Promise.resolve());
+  //  flatten array of promises
+  var promises = [];
+  files.forEach(function(sassFile) {
+    promises = promises.concat(RSVP.all(self.compileSassFile(srcPath, sassFile, destDir)));
+  });
+  // throttle number of calls to compileSassFile
+  var numConcurrentCalls = os.cpus().length;
+  var promiseChunks = [];
+  for (var index = 0; index < promises.length; index += numConcurrentCalls) {
+    var chunk = promises.slice(index, index + numConcurrentCalls);
+    promiseChunks.push(chunk);
+  }
+ return promiseChunks.reduce(function(promise, chunk) {
+   return promise.then(function() {
+     return RSVP.all(chunk);
```

Comment:
Thank you, I am checking in with Michael about this and will give it a try tonight.